File: frontend/src/types/employee.ts
export interface Country {
  id: number;
  name: string;
  code: string;
  currency: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface Department {
  id: number;
  name: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface JobTitle {
  id: number;
  name: string;
  department_id: number;
  department?: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface Employee {
  id: number;
  first_name: string;
  last_name: string;
  email: string;
  job_title_id: number;
  country_id: number;
  salary: number;
  address?: string;
  join_date: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;

  // Denormalized fields populated by the backend via JOINs
  job_title?: string;
  department?: string;
  country?: string;
  currency?: string;
}

export interface EmployeeFormData {
  first_name: string;
  last_name: string;
  email: string;
  job_title_id: number;
  country_id: number;
  salary: number;
  address: string;
  join_date: string;
}

export interface EmployeeListResult {
  employees: Employee[];
  total: number;
  page: number;
  limit: number;
}

export interface EmployeeFilter {
  search?: string;
  country_id?: number;
  job_title_id?: number;
  department_id?: number;
  page: number;
  limit: number;
}

export interface SalaryRange {
  country: string;
  min: number;
  max: number;
  average: number;
  median: number;
  count: number;
}

export interface SalaryByTitle {
  country: string;
  job_title: string;
  average: number;
  min: number;
  max: number;
  count: number;
}

export interface DepartmentStats {
  department: string;
  average_salary: number;
  min_salary: number;
  max_salary: number;
  employee_count: number;
}

export interface CountryHeadcount {
  country: string;
  employee_count: number;
  average_salary: number;
}

export interface OrgSummary {
  total_employees: number;
  average_salary: number;
  total_countries: number;
  total_departments: number;
  country_breakdown: CountryHeadcount[];
}

File: backend/internal/model/employee.go
package model

import "time"

type Employee struct {
	ID           int64     `json:"id"`
	FirstName    string    `json:"first_name"`
	LastName     string    `json:"last_name"`
	Email        string    `json:"email"`
	JobTitleID   int64     `json:"job_title_id"`
	CountryID    int64     `json:"country_id"`
	Salary       float64   `json:"salary"`
	Address      string    `json:"address,omitempty"`
	JoinDate     time.Time `json:"join_date"`
	IsActive     bool      `json:"is_active"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`

	// Denormalized fields populated via JOINs (not persisted on Employee row)
	JobTitle   string `json:"job_title,omitempty"`
	Department string `json:"department,omitempty"`
	Country    string `json:"country,omitempty"`
	Currency   string `json:"currency,omitempty"`
}

func (e Employee) FullName() string {
	if e.FirstName == "" {
		return e.LastName
	}
	if e.LastName == "" {
		return e.FirstName
	}
	return e.FirstName + " " + e.LastName
}

// EmployeeFilter is the input to EmployeeRepository.List. It combines
// query filters with pagination (offset/limit, embedded). A zero or negative
// Limit means "return all matching records".
type EmployeeFilter struct {
	Search       string `json:"search"`
	CountryID    int64  `json:"country_id"`
	JobTitleID   int64  `json:"job_title_id"`
	DepartmentID int64  `json:"department_id"`
	Pagination
}

// EmployeeListResult is a page of employees plus the unfiltered total.
type EmployeeListResult struct {
	Employees []Employee `json:"employees"`
	Total     int64      `json:"total"`
	Limit     int        `json:"limit"`
	Offset    int        `json:"offset"`
}

type SalaryRange struct {
	Country string  `json:"country"`
	Min     float64 `json:"min"`
	Max     float64 `json:"max"`
	Average float64 `json:"average"`
	Median  float64 `json:"median"`
	Count   int64   `json:"count"`
}

type SalaryByTitle struct {
	Country  string  `json:"country"`
	JobTitle string  `json:"job_title"`
	Average  float64 `json:"average"`
	Min      float64 `json:"min"`
	Max      float64 `json:"max"`
	Count    int64   `json:"count"`
}

type DepartmentStats struct {
	Department    string  `json:"department"`
	AverageSalary float64 `json:"average_salary"`
	MinSalary     float64 `json:"min_salary"`
	MaxSalary     float64 `json:"max_salary"`
	EmployeeCount int64   `json:"employee_count"`
}

type CountryHeadcount struct {
	Country       string  `json:"country"`
	EmployeeCount int64   `json:"employee_count"`
	AverageSalary float64 `json:"average_salary"`
}

type OrgSummary struct {
	TotalEmployees   int64              `json:"total_employees"`
	AverageSalary    float64            `json:"average_salary"`
	TotalCountries   int64              `json:"total_countries"`
	TotalDepartments int64              `json:"total_departments"`
	CountryBreakdown []CountryHeadcount `json:"country_breakdown"`
}
